pkg/controller/daemon/daemoncontroller.go

```diff
@@ -818,9 +822,8 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 			// unexpected
 			case
 				predicates.ErrPodAffinityNotMatch,
-				predicates.ErrServiceAffinityViolated,
-				predicates.ErrTaintsTolerationsNotMatch:
-				return false, false, false, fmt.Errorf("unexpected reason: GeneralPredicates should not return reason %s", reason.GetReason())
+				predicates.ErrServiceAffinityViolated:
+				return false, false, false, fmt.Errorf("unexpected reason: deamonSetPredicates should not return reason %s", reason.GetReason())
```

Comment:
All reasons are logged at the beginning:

```go
for _, r := range reasons {
	glog.V(4).Infof("daemonSetPredicates failed on ds '%s/%s' for reason: %v", ds.ObjectMeta.Namespace, ds.ObjectMeta.Name, r.GetReason())
	switch reason := r.(type) {
...
```

Are you suggesting we log unexpected predicate errors (probably with `glog.Warningf`?) in addition to this log?